File: two-pointers/isPalindrome.py
class Solution(object):
    def isPalindrome(self, s):
        """
        :type s: str
        :rtype: bool
        """
        left = 0
        right = len(s)-1
        while left<= right:
            while left<right and not self.alphaNum(s[left]):
                left+=1
            while left<right and not self.alphaNum(s[right]):
                right-=1
            if s[left].lower()!=s[right].lower():
                return False
            left+=1
            right-=1
        return True
    def alphaNum(self, c):
        return ord('A') <= ord(c) <= ord('Z') or ord('0') <= ord(c) <= ord('9') or ord('a') <= ord(c) <= ord('z')
s = "race a car"
s = "A man, a plan, a canal: Panama"
print(Solution().isPalindrome(s))

File: two-pointers/isPalindrome.go
package main

import (
	"fmt"
	"strings"
)

func isPalindrome(s string) bool {
	l, r := 0, len(s)-1
	for l < r {
		for l < r && !isAlphaNumeric(int(s[l])) {
			l += 1
		}
		for l < r && !isAlphaNumeric(int(s[r])) {
			r -= 1
		}
		if strings.ToLower(string(s[l])) != strings.ToLower(string(s[r])) {
			return false

		}
		l += 1
		r -= 1

	}
	return true

}
func isAlphaNumeric(asciCode int) bool {
	return (int('a') <= asciCode && asciCode <= int('z') ||
		int('A') <= asciCode && asciCode <= int('Z') ||
		int('0') <= asciCode && asciCode <= int('9'))
}
func main() {
	fmt.Println(isPalindrome("race a car"))

}
